pynet/datasets/hbn.py

```diff
@@ -0,0 +1,733 @@
+# -*- coding: utf-8 -*-
+########################################################################
+# NSAp - Copyright (C) CEA, 2019
+# Distributed under the terms of the CeCILL-B license, as published by
+# the CEA-CNRS-INRIA. Refer to the LICENSE file or to
+# http://www.cecill.info/licences/Licence_CeCILL-B_V1-en.html
+# for details.
+########################################################################
+
+"""
+Module provides functions to prepare different datasets from HBN.
+"""
+
+# Imports
+import os
+import json
+import time
+import urllib
+import shutil
+import pickle
+import requests
+import logging
+import numpy as np
+from collections import namedtuple
+import pandas as pd
+import sklearn
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import StandardScaler, OneHotEncoder
+from sklearn.linear_model import LinearRegression
+from pynet.datasets import Fetchers
+from neurocombat_sklearn import CombatModel as fortin_combat
+from nilearn.surface import load_surf_data
+
```

Comment:
Can we factorize here?
I have the feeling that the path to the data are different and maybe also some variables but the code is the same.

Reply:
Do you want me to factorize the whole code? It is indeed redundant

Reply:
Yes. Just import the base functions from the euaims fetcher.

Reply:
Done